packages/mui-joy/src/styles/variantUtils.ts

```diff
@@ -420,3 +450,71 @@ export const createSolidInversion = (theme: ThemeFragment, addDefaultValues?: bo
   });
   return result;
 };
+
+/**
+ *
+ * @param color a supported theme color palette
+ * @returns (theme: ThemeFragment) => Record<DefaultColorPalette, CSSObject>
+ */
+export const applySolidInversion = (color: ColorPaletteProp) => (theme: ThemeFragment) => {
```

Comment:
Do we plan to expose 4 different methods? Can't we have one where we would accept the variant & the color?

Reply:
> Do we plan to expose 4 different methods? Can't we have one where we would accept the variant & the color?

We need 2 functions so far since only `soft` and `solid` come with a background. I think a separate function sounds better in the long term.